helm-chart/binderhub/templates/deployment.yaml

```diff
@@ -88,10 +88,11 @@ spec:
             secretKeyRef:
               name: binder-secret
               key: "binder.hub-token"
-        {{- if or (ne .Values.jupyterhub.auth.type "custom") (ne .Values.jupyterhub.auth.custom.className "nullauthenticator.NullAuthenticator") }}
-        # auth is enabled
+        {{- if .Values.config.BinderHub.auth_enabled }}
+        - name: JUPYTERHUB_API_URL
+          value: {{ (print (.Values.config.BinderHub.hub_url | trimSuffix "/") "/hub/api/") }}
         - name: JUPYTERHUB_SERVICE_PREFIX
-          value: {{ .Values.baseUrl | quote }}
+          value: {{ .Values.config.BinderHub.base_url | quote }}
```

Comment:
Note: this will usually be null. Is that okay?

Reply:
I think perhaps instead, this should be moved to binderhub_config.py in the image which has more information available, rather than environment variables.

Reply:
do you mean to have `c.HubOAuth.base_url = get_value('config.BinderHub.base_url ', '/')` in binderhub_config.py?

Reply:
`.Values.config.BinderHub.base_url` can definitely be null. It is undefined by default.

This is the difference between setting these things in configuration vs setting them in the chart.

`BinderHub.base_url` on the *instantiated object* cannot be null. But that doesn't mean that it must be specified either in the chart or in 

So, instead, we can gate this with:

```yaml
{{- $default "/" .Values.config.BinderHub.base_url }}
...
```

But I think it would still be better to do that in the image than in the helm env.

Reply:
@minrk I pushed a new commit. But I am not sure if this is what you wanted.